docs/gcp_setup.md

```diff
@@ -0,0 +1,287 @@
+# Google Cloud Platform Setup for a BigQuery Target
+
+## Overview
+
+This page details Google Cloud components required, with recommended minimal privileges, to use GOE in your GCP project.
+
+### Service Account
+
+A service account should be provisioned from the GCP project. This service account can be used by any service that will execute GOE commands, for example it could be attached to a GCE virtual machine.
+
+### Cloud Storage Bucket
+
+A cloud storage bucket is required to stage data before ingesting it into BigQuery. Ensure the bucket is in a location compatible with the target BigQuery dataset.
+
+### Dataproc (Spark)
+
+For non-trivially sized tables GOE uses Spark to copy data from the source database to cloud storage. In a GCP setting this is likely to be provided by one of two services:
+
+1. Dataproc Batches
+1. Dataproc
+
+### Roles
+
+The role names below are used throughput this page but can be changed to suit company policies. These roles will provide adequate access to stage data in cloud storage and load it into BigQuery.
+
+| Role                | Mandatory | Purpose                                                                         |
+| ------------------- | ----------| ------------------------------------------------------------------------------- |
+| `goe_gcs_role`      |      Y    | Permissions to read/write on the GOE staging bucket.                            |
+| `goe_bq_core_role`  |      Y    | Core permissions to interact with BigQuery, list datsets/tables/etc.<br />No data read/write permissions.<br />Will be granted at the project level. |
+| `goe_bq_app_role`   |      Y    | Permissions to read/write data in the final dataset.<br />Optionally can include table create/drop permissions.<br />Locked down at dataset level. |
+| `goe_bq_stg_role`   |      Y    | Permissions to read data and create/drop staging tables in the staging dataset.<br />Locked down at dataset level. |
+| `goe_dataproc_role` |      N    | Permissions to interact with a permanent Dataproc cluster.                      |
+| `goe_batches_role`  |      N    | Permissions to interact with Dataproc Batches service.                          |
+
+### Compute Engine Virtual Machine
+
+To work interactively with GOE you need to be able run commands with appropriate permissions. Rather than downloading service account keys we believe it is better to attach the service account to a GCE virtual machine and run all commands from there.
+
+The virtual machine does not need to be heavily resourced; most of the heavy lifting is done by Spark and BigQuery.
```

Comment:
The virtual machine requires limited resources as most of the intensive work is done by Spark and BigQuery.